src/model/encoding/convertFromHTMLToContentBlocks.js

```diff
@@ -367,12 +375,22 @@ function genFragment(
   }
 
   var entityId: ?string = null;
-  var href: ?string = null;
 
   while (child) {
-    if (nodeName === 'a' && child.href && hasValidLinkText(child)) {
-      href = new URI(child.href).toString();
-      entityId = DraftEntity.create('LINK', 'MUTABLE', {url: href});
+    if (nodeName === 'a' && child instanceof Element && child.href && hasValidLinkText(child)) {
```

Comment:
Managed to remove the Object casting but still need to do an Element assignment in order to have access to `getAttribute()` without failling on flow

since if trying to access property anchor[attr] directly causes the error:

```
access of computed property/element. Indexable signature not found in
```